### ISSUE TITLE
Relax DOC rules for private/inner methods

### ISSUE THREAD

**Issue by cbornet** [astral-sh/ruff#20280]
### Summary

Currently the following code triggers DOC201 (missing return)

```python
def _my_helper_function() -> str
   """This is some helper function."""
    return "foo"
```

Whereas this doesn't trigger:

```python
def _my_helper_function() -> str
    return "foo"
```

It would be nice if we could put just a short description to the method without a full-blown docstring to private and inner functions/methods.

**Comment by cbornet** [comment]
It is like automatically setting `ignore-one-line-docstrings = true` for non-visible methods.

**Comment by ntBre** [comment]
This makes some sense to me, maybe as a separate configuration option. The `pydocstyle` (`D`) rules differentiate between private and public functions, but the `pydoclint` (`DOC`) rules don't.
